### PR TITLE
NNVP-39: API POST registrations

### PR DESCRIPTION
**Pull request by eryk7b**

### CODE REVIEW

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-15** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-15.cluster.development.nnvp.thegluesandbox.com)

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-15** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-15.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/controller/RsrCheckRestController.java [hunk 1]
```diff
@@ -55,4 +60,23 @@ public RsrCheckRestController(MapperProvider mapperProvider,
...
+  public ResponseEntity<SuccessResponseDTO<RegistrationCreatedResponseDTO>> createRegistration(
+      @Valid @RequestBody RegistrationSpecificationDTO registration) {
+    validator.validate(registration);
+    return new ResponseEntity<>(
+        SuccessResponseDTO.successResponse(
+            RegistrationCreatedResponseDTO.builder()
+                .registrationCreated(new RegistrationCreatedDTO(
+                    rsrCheckService.createRegistration(registration)
+                        .toString()
+                ))
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/RsrCheckRestController.java]: lets follow accento and just return RegistrationSpecificationDTO in response

File: src/main/java/com/theglue/nnvp/controller/dto/RegistrationCreatedDTO.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class RegistrationCreatedDTO {
+
+  @ApiModelProperty(value = "Created entity id")
+  private String id;
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/dto/RegistrationCreatedDTO.java]: why is it a String if entity id is integer

File: src/main/java/com/theglue/nnvp/controller/dto/request/RegistrationSpecificationDTO.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.theglue.nnvp.controller.dto.request;
+
+import io.swagger.annotations.ApiParam;
+import lombok.Data;
+
+import javax.validation.constraints.NotNull;
+
+@Data
+public class RegistrationSpecificationDTO extends RsrCheckRequest {
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/dto/request/RegistrationSpecificationDTO.java]: I wouldn't use inheritance here. These 2 objects shere a lot of fields only by accident.
The don't seem to be logically connected.

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-15** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-15.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**